crates/client/src/client.rs

```diff
@@ -745,9 +745,12 @@ impl FuelClient {
         Ok(block)
     }
 
-    pub async fn block_by_height(&self, height: u32) -> io::Result<Option<types::Block>> {
+    pub async fn block_by_height(
```

Comment:
The types on `latest_gas_price` include a `u32`  to represent block height. Should we use `BlockHeight` there?